File: fila/Fila Estática/Java/Fila/FilaEstatica.java
public class FilaEstatica implements Enfileiravel {
	//variaveis de instância/globais
	private int ponteiroInicio;
	private int ponteiroFim;
	private Object[] dados;

	//construtores
	public FilaEstatica(int tamanho) {
		ponteiroInicio = 0;
		ponteiroFim = -1;
		dados = new Object[tamanho];
	}

	public FilaEstatica() {
		this(10);
	}

	//métodos principais
	@Override
	public void enfileirar(Object dado){
		if (!estaCheia()) {
			ponteiroFim++;
			dados[ponteiroFim] = dado;
		} else {
			System.err.println("Fila Cheia!");
		}
	}

	@Override	
	public Object desenfileirar(){
		Object dadoInicio = null;
		if (!estaVazia()) {
			dadoInicio = dados[ponteiroInicio];
			ponteiroInicio++;
		} else {
			System.err.println("Fila Vazia!");
		}
		return dadoInicio;		
	}
	
	@Override	
	public Object espiar(){
		Object dadoInicio = null;
		if (!estaVazia()) {
			dadoInicio = dados[ponteiroInicio];
		} else {
			System.err.println("Fila Vazia!");		
		}
		return dadoInicio;
	}
	
	//métodos auxiliares	
	@Override	
	public boolean estaCheia(){
		return (ponteiroFim == dados.length - 1);
	}

	@Override	
	public boolean estaVazia(){
		return (ponteiroInicio == ponteiroFim + 1);
	}

	@Override	
	public String imprimir(){
		String resultado = "[";
		for (int i = ponteiroInicio; i <= ponteiroFim; i++) {
			//resultado += dados[i]+ " ";
			if (i == ponteiroFim) {
				resultado += dados[i];
			} else {
				resultado += dados[i] + ",";
			}
		}
		return resultado + "]";		
	}
}

File: fila/Fila Estática/Kotlin/Fila/FilaEstatica.kt
class FilaEstatica(private val tamanho: Int = 10) : Enfileiravel {
 
	private var ponteiroInicio = 0
	private var ponteiroFim = -1 
	private var dados: Array<Any?> = arrayOfNulls(tamanho)
    
	override fun enfileirar(dado: Any?) {
		if (!estaCheia()) {
			ponteiroFim++
			dados[ponteiroFim] = dado
		} else {
			println("Queue is full!")
		}
	}
	
	override fun desenfileirar(): Any? {
		var dadoInicio: Any? = null
		if (!estaVazia()) {
			dadoInicio = dados[ponteiroInicio]
			ponteiroInicio++
		} else {
			println("Queue is empty!")
		}
		return dadoInicio
	}
	
	override fun espiar(): Any? {
		var dadoInicio: Any? = null
		if (!estaVazia()) {
			dadoInicio = dados[ponteiroInicio]
		} else {
			println("Queue is empty!")
		}
		return dadoInicio
	}
	
	override fun estaCheia(): Boolean {
		return ponteiroFim == dados.size - 1
	}
	
	override fun estaVazia(): Boolean {
		return ponteiroInicio == ponteiroFim + 1
	}
	
	override fun imprimir(): String {
		var resultado = "["
		for (i in ponteiroInicio..ponteiroFim) {
			if (i == ponteiroFim)
				resultado += "${dados[i]}"
			else
				resultado += "${dados[i]},"
		}
		return "$resultado]"
	}
}